critiquebrainz/frontend/views/review.py

```diff
@@ -226,12 +226,17 @@ def create():
         return redirect(url_for('user.reviews', user_id=current_user.id))
 
     # Checking if the user already wrote a review for this entity
-    reviews, count = db_review.list_reviews(user_id=current_user.id, entity_id=entity_id)
+    reviews, count = db_review.list_reviews(user_id=current_user.id, entity_id=entity_id, inc_drafts=True, inc_hidden=True)
     review = reviews[0] if count != 0 else None
 
     if review:
-        flash.error(gettext("You have already published a review for this entity!"))
-        return redirect(url_for('review.entity', id=review["id"]))
+        if review['is_draft']:
+            return redirect(url_for('review.edit', id=review['id']))
+        elif review['is_hidden']:
+            return redirect(url_for('review.entity', id=review['id']))
```

Comment:
What does this mean if a user tries to write a review for an item that has been hidden by an admin?
It means that they'll try and go to (for example) /write/artist/[id], which will redirect them to /review/[reviewid], which will then raise this alert, right? https://github.com/metabrainz/critiquebrainz/blob/159d934ff7886ab33cd72e89f6e1cf0f26fe9999/critiquebrainz/frontend/views/review.py#L77-L81

Reply:
AFAIU, items are not hidden by admin but rather `review`s are. So, the case added above is executed when a person wrote a review for some item which was subsequently hidden by the admin and then they try to write a review again. This redirect will take them to the review page where they'll be shown the warning you linked.